.github/workflows/deploy-prerelease.yml

```
@@ -0,0 +1,224 @@
+  group: deploy-development
+  cancel-in-progress: true
+
+jobs:
+  check-dispatch-token:
+    name: Check dispatch token
+    runs-on: ubuntu-latest
+    if: ${{ github.event_name == 'repository_dispatch' }}
+    steps:
+      - name: Check token
+        run: |
+          if [[ ${{ github.event.client_payload.token}} != ${{secrets.REPO_ACCESS_TOKEN }} ]]; then
+            echo "Invalid token"
+            exit 1
+          fi
+
+  get-deployment-sites:
+    name: Get Deployment Sites
+    runs-on: ubuntu-latest
+    needs: check-dispatch-token
+    # Don't run if the event is triggered by a repository dispatch with a non-valid token, 
+    # but trigget if the event is not a repository dispatch
+    if: ${{ !cancelled() && !(github.event_name == 'repository_dispatch' && needs.check-dispatch-token.result != 'success') }}
+    outputs:
+      deployment-sites: ${{ steps.get-deployment-sites.outputs.deployment-sites }}
+    steps:
+      - name: Checkout config
+        uses: actions/checkout@v4
+
+      - name: Get sites
+        id: get-deployment-sites
+        run: echo "deployment-sites=$(jq --compact-output '.sites' ./config/deployment-sites.json)" >> $GITHUB_OUTPUT
+
+  pack:
+    name: Pack environment
+    runs-on: ubuntu-latest
+    needs: check-dispatch-token
+    # Don't run if the event is triggered by a repository dispatch with a non-valid token, 
+    # but trigget if the event is not a repository dispatch
+    if: ${{ !cancelled() && !(github.event_name == 'repository_dispatch' && needs.check-dispatch-token.result != 'success') }}
+    outputs:
+      full-name: ${{ steps.get-env-name.outputs.full-name }}
+      dependecy-names: ${{ steps.get-dependencies.outputs.dependecy-names }}
+      pr-numbers: ${{ steps.get-dependencies.outputs.pr-numbers }}
+    env:
+      GH_TOKEN: ${{ github.token }}
+      DEPENDENCY_REPOS: replace_landsurface era5_grib_parallel
+      ARCH: noarch
+      CHANNEL_SUFFIX: _channel
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Get environment name
+        id: get-env-name
+        run: echo "full-name=$(yq '.name' < ${{env.ENV_NAME}})-${{env.VERSION}}" >> $GITHUB_OUTPUT
+
+      # Search for the latest open PR (not a draft) in the replace_landsurface and era5_grib_parallel repos with
+      # changes in specific files. If a PR is found, download the artifact from the PR latest 
+      # (successful) workflow run. If a PR is not found, download the lates released version.
+      - name: Get latest dependencies
+        id: get-dependencies
+        run: |
+          download_latest_release() {
+            gh release download $(gh release view --repo ${{ github.repository_owner }}/$1 --json tagName --jq '.tagName') \
+            --repo ${{ github.repository_owner }}/$1 --pattern *.tar.bz2 -D ${1}${{env.CHANNEL_SUFFIX}}/${{env.ARCH}} && \
+            echo "${1}: Successfully downloaded latest release: \"$(basename ${1}${{env.CHANNEL_SUFFIX}}/${{env.ARCH}}/*.tar.bz2)\"."
```

Can this be split apart a bit? Like can you do the download and then check the exit code as separate commands rather than doing `gh download ... && echo "Success"`? It might be a bit less to keep in the brain while reading this. 
 ```suggestion
            gh release download $(gh release view --repo ${{ github.repository_owner }}/$1 --json tagName --jq '.tagName') \
              --repo ${{ github.repository_owner }}/$1 --pattern *.tar.bz2 -D ${1}${{env.CHANNEL_SUFFIX}}/${{env.ARCH}}
            
            if [[ "$?" == "0" ]]; then
              echo "${1}: Successfully downloaded latest release: \"$(basename ${1}${{env.CHANNEL_SUFFIX}}/${{env.ARCH}}/*.tar.bz2)\"."
            fi
```

---

Fixed in <COMMIT>